src/schema/schema.rs

```diff
@@ -516,12 +538,13 @@ mod tests {
             let json_err = schema.parse_document(r#"{
                 "title": "my title",
                 "author": "fulmicoton",
-                "count": 50000000000000000000
+                "count": 50,
+                "popularity": 9223372036854775808,
             }"#);
             match json_err {
-                Err(NotJSON(_)) => {
+                Err(NotJSON(_)) | Err(NotJSONObject(_)) => {
```

Comment:
Hmm yeh that's not great, is it. Maybe those errors just need better names, or could be combined into 1